# LECTURA DE DATASET E IMPORTACION DE LIBRERIAS

In [2]:
#librerías de manejo de datos
import numpy as np
import pandas as pd

#librerías de visualización de datos
import matplotlib.pyplot as plt
import seaborn as sns


import subprocess
import sys

try:
    import folium
except ImportError as e:
    #%pip install folium wget
    subprocess.check_call([sys.executable, "-m", "pip", "install", 'folium'])
finally:
    import folium

from folium.plugins import MarkerCluster
from folium.plugins import MousePosition
from folium.features import DivIcon

# Construye la ruta relativa al archivo
ruta_relativa = "../primer_proyecto_datos/milan_airbnb.csv"

# Lee el archivo CSV
df = pd.read_csv(ruta_relativa)

In [3]:
df.head(5)

id                                name  host_id    host_name  \
0   6400                    The Studio Milan    13822    Francesca   
1  23986     " Characteristic Milanese flat"    95941       Jeremy   
2  28300             nice flat near the park   121663        Marta   
3  32119  Nico & Cynthia's Easy Yellow Suite   138683  Nico&Cinzia   
4  32649       Nico&Cinzia's Red Easy Suite!   138683  Nico&Cinzia   

  neighbourhood  latitude  longitude        room_type  price  minimum_nights  \
0       TIBALDI  45.44119    9.17813     Private room    100               4   
1       NAVIGLI  45.44806    9.17373  Entire home/apt    150               1   
2         SARPI  45.47647    9.17359     Private room    180               1   
3   VIALE MONZA  45.52014    9.22300  Entire home/apt     75               2   
4   VIALE MONZA  45.51874    9.22495  Entire home/apt     71               2   

   number_of_reviews last_review  reviews_per_month  \
0                 12    19/04/10               0.14   
1                 15    07/09/20               0.21   
2                  8    22/04/12               0.11   
3                 15    01/07/18               0.23   
4                 29    23/10/16               0.71   

   calculated_host_listings_count  availability_365  
0                               1               358  
1                               1               363  
2                               1               365  
3                               3               200  
4                               3               308

In [4]:
# Function que facilita la exploración básica del dataset
def summary(df):
    total_values = df.shape[0]
    missing_data = df.isnull().sum()
    missing_percentage = (missing_data / total_values) * 100
    print(f"Dataset has {df.shape[1]} features and {df.shape[0]} rows.")    
    summary = pd.DataFrame(index=df.columns)
    summary["Unique"] = df.nunique().values
    summary["Missing"] = df.isnull().sum().values
    summary['Missing %'] = ((missing_data / total_values) * 100).round(2)
    summary["Duplicated"] = df.duplicated().sum()
    summary["Types"] = df.dtypes
    return summary

In [5]:
summary(df)

Dataset has 15 features and 18322 rows.


Unique  Missing  Missing %  Duplicated  \
id                               18322        0       0.00           0   
name                             18050       10       0.05           0   
host_id                          12213        0       0.00           0   
host_name                         2917      124       0.68           0   
neighbourhood                       87        0       0.00           0   
latitude                          7419        0       0.00           0   
longitude                         9134        0       0.00           0   
room_type                            4        0       0.00           0   
price                              465        0       0.00           0   
minimum_nights                      68        0       0.00           0   
number_of_reviews                  421        0       0.00           0   
last_review                       2039     5062      27.63           0   
reviews_per_month                  648     5062      27.63           0   
calculated_host_listings_count      49        0       0.00           0   
availability_365                   366        0       0.00           0   

                                  Types  
id                                int64  
name                             object  
host_id                           int64  
host_name                        object  
neighbourhood                    object  
latitude                        float64  
longitude                       float64  
room_type                        object  
price                             int64  
minimum_nights                    int64  
number_of_reviews                 int64  
last_review                      object  
reviews_per_month               float64  
calculated_host_listings_count    int64  
availability_365                  int64

In [7]:
df.describe()

id       host_id      latitude     longitude         price  \
count  1.832200e+04  1.832200e+04  18322.000000  18322.000000  18322.000000   
mean   2.587132e+07  8.494918e+07     45.471318      9.187382    115.094913   
std    1.525744e+07  1.040579e+08      0.020731      0.029543    290.793019   
min    6.400000e+03  1.944000e+03     45.395050      9.060680      8.000000   
25%    1.214067e+07  1.243642e+07     45.454462      9.168782     50.000000   
50%    2.588718e+07  3.141304e+07     45.470875      9.186291     73.500000   
75%    3.926758e+07  1.335227e+08     45.486850      9.209680    110.000000   
max    5.095505e+07  4.117208e+08     45.539850      9.275280  11999.000000   

       minimum_nights  number_of_reviews  reviews_per_month  \
count    18322.000000       18322.000000       13260.000000   
mean         5.798112          23.720827           0.799344   
std         26.687720          57.657486           1.220346   
min          1.000000           0.000000           0.010000   
25%          1.000000           0.000000           0.100000   
50%          2.000000           3.000000           0.310000   
75%          3.000000          19.000000           0.990000   
max       1124.000000         896.000000          13.710000   

       calculated_host_listings_count  availability_365  
count                    18322.000000      18322.000000  
mean                        14.421897        153.508624  
std                         46.295635        138.757302  
min                          1.000000          0.000000  
25%                          1.000000          0.000000  
50%                          1.000000        123.000000  
75%                          4.000000        302.000000  
max                        314.000000        365.000000

## Resumen Estadístico General

**- ¿Cuántas filas y columnas tiene el dataset?** Dataset has 15 features and 18322 rows.
**- ¿Cuál es el rango de valores de las variables numericas** 
1. latitude: min (45.395050) max (45.539850)
2. longitude: min (9.060680) max (9.275280)
3. price: min (8.000000) max (11999.000000)
4. minimum_nights: min (1.000000) max (1124.000000)
5. number_of_reviews: min (0.000000) max (896.000000)
6. reviews_per_month: min (0.010000) max (13.710000)
7. calculated_host_listings_count: min (1.000000) max (314.000000)
8. availability_365: min (0.000000) max (365.000000)

**-¿Cuántos valores únicos hay en las variables categóricas (name, host_name, neighbourhood, room_type)?
- name	18050
- host_name	2917
- neighbourhood	87
- room_type	4


¿Cuántos registros tienen valores faltantes?
- name	 10 null  --> 0.05%
- host_name	 124 null 
- last_review	5062 --> 27.63%
- reviews_per_month  5062  --> 27.63%